### DISCUSSION TITLE
Support a popover like behavior for tooltips

### DISCUSSION THREAD

**Discussion by OgDev-01** [plouc/nivo#2750] [Ideas]
**Is your feature request related to a problem? Please describe.**

Currently working on a project and making use of the scatterplot chart. But discovered some limitations with the tooltip provided

- Tooltip can't be hovered on

https://user-images.githubusercontent.com/62995161/207777072-97967a39-3b3c-4507-b978-2164cc2df77c.mov

**Describe the solution you'd like**

A behavior similar to GitHub hovercard where we can hiver on the tooltip itself

https://user-images.githubusercontent.com/62995161/207776852-7d605c29-e78c-459f-b99f-f5a1b83e2fde.mov

**Describe alternatives you've considered**

**Additional context**

**Comment by bdougie** [comment]
Bump

**Comment by takanome-dev** [comment]
cc @plouc can [this](https://github.com/plouc/nivo/issues/2201#issue-1497795189) be achieved? Would be very happy to help with this if possible :smiley:

**Comment by plouc** [comment]
I think it's more the behavior of a popover, but it's definitely possible to implement this in the lib.

However `@nivo/tooltip` is a core component used for all charts, and we currently assume that the tooltip disappear on mouse leave, meaning that we would need to change all charts' event handlers.

@takanome-dev, if you want to give it a try, that would be great indeed :)

**Comment by plouc** [comment]
We would also need to introduce a new property to control this behavior.

**Comment by takanome-dev** [comment]
> I think it's more the behavior of a popover, but it's definitely possible to implement this in the lib.
> 
> However `@nivo/tooltip` is a core component used for all charts, and we currently assume that the tooltip disappear on mouse leave, meaning that we would need to change all charts' event handlers.
> 
> @takanome-dev, if you want to give it a try, that would be great indeed :)

That sounds a lot :sweat_smile: Alright, I'm gonna give it a shot :+1: 

> We would also need to introduce a new property to control this behavior.

Can you elaborate more on that?

**Comment by plouc** [comment]
I mean that it might not be the expected behavior, tooltip VS popover, so for retro-compatibility, this should be an opt-in IMHO.

**Comment by plouc** [comment]
I think that a reasonable approach would be to start with a single chart (maybe the bar chart), before going all in and changing the numerous packages that nivo provides 😅.

**Comment by takanome-dev** [comment]
I agree with that :sweat_smile: 
<img src="https://media1.giphy.com/media/5xtDarIN81U0KvlnzKo/giphy.gif"/>

**Comment by plouc** [comment]
Also using a property and keeping the current behavior by default would allow us to gradually ship this feature.

**Comment by plouc** [comment]
I was having a look at the open sauced dashboard, and I understand why you want such feature 😄

**Comment by takanome-dev** [comment]
Yeah it would be pretty cool to have it 😃

**Comment by plouc** [comment]
Seems like a nice project btw, would you mind if I add it to the references?

**Comment by takanome-dev** [comment]
> Seems like a nice project btw, would you mind if I add it to the references?

That would be great 😊 Thoughts @bdougie?

**Comment by plouc** [comment]
The references page is not that nice right now, just a list of bare links, but I plan to change that to include some screenshots, maybe something similar to https://nextjs.org/showcase or https://www.gatsbyjs.com/customers/.

**Comment by bdougie** [comment]
> Seems like a nice project btw, would you mind if I add it to the references?

Yes, please do. Also thanks for the engaging discussion.

**Comment by plouc** [comment]
Thank you @bdougie and @takanome-dev!

**Comment by jasongerbes** [comment]
Has this idea been abandoned? I have a similar use case

**Comment by bdougie** [comment]
We actually figured out a work around in this PR https://github.com/open-sauced/app/pull/1266

@OgDev-01 can probably provide more context here.

**Comment by plouc** [comment]
I'm moving this to a discussion, as it's not really an issue, it's possible to bypass this limitation for certain chart types using a custom component, and extra layer, or custom mouse event handlers.
